src/Components/RankingSection.tsx

```diff
@@ -193,9 +193,32 @@ const MoreBtn = styled.button`
   }
 `
 
+const FILTER_KEY = 'ranking_selected_filter'
+const TAB_KEY = 'ranking_selected_tab'
+
 const RankingSection = () => {
-  const [selectedFilter, setSelectedFilter] = useState<FilterKey>('all')
-  const [selectedTab, setSelectedTab] = useState<TabKey>('want')
+  // localStorage에서 초기값 불러오기
+  const getInitialFilter = () => {
+    const saved = localStorage.getItem(FILTER_KEY)
+    if (saved === 'all' || saved === 'female' || saved === 'male' || saved === 'teen') return saved as FilterKey

```

Comment:
```suggestion
const rankingGroup = ['all','femail','mail','teen'];

const isRankingGroup = (string:string) : string is FilterKey => rankingGroup.includes(string);

if (saved && isRankingGroup(saved)) return saved;
```
타입 가드를 통해 타입의 안정성을 확보할 수 있을거 같아요! 다중 조건문은 배열로 선언해서 includes로 개선할 수 있을거 같아요!